core/src/main/java/org/aerogear/mobile/core/http/OkHttpResponse.java

```diff
@@ -90,8 +101,9 @@ public HttpResponse onError(Runnable errorHandler) {
     @Override
     public HttpResponse onSuccess(Runnable successHandler) {
         this.successHandler = successHandler;
-        // If there is _any_ response the success handler should run
-        if (response != null) {
+        // If there is _any_ response and it is successful the success handler should run
+        // immediately
+        if (response != null && (response.isSuccessful() || response.isRedirect())) {
             runSuccessHandler();
```

Comment:
As the success handler is immediately called upon setting a success handler using `onSuccess`, I needed to adjust the condition to run it. In some cases, I saw the response is received faster than execution arrives at setting `onSuccess` handler.